### ISSUE TITLE
The LG.arcs_as_tensor().shape becomes extreme small after compose with ctc_topo

### ISSUE THREAD

**Issue by Jarvan-Wang** [k2-fsa/snowfall#159]
The env I use (I know it's not the latest commit, but I had ran the librispeech exp and it's ok):
k2:
    k2-fsa/k2@9c3c1f64543b246af3c9e7288cfd01677954058c
snowfall:
    k2-fsa/snowfall@bbcf1ec1185179b9b502287c51b46fbc2accb8d0

Here is my own exp code:
```
#phones.txt and words.txt is from my ASR of kaldi
first_phone_disambig_id = find_first_disambig_symbol(phone_symbol_table)
first_word_disambig_id = find_first_disambig_symbol(word_symbol_table)
phone_ids = get_phone_symbols(phone_symbol_table)
phone_ids_with_blank = [0] + phone_ids
ctc_topo = k2.arc_sort(build_ctc_topo(phone_ids_with_blank))
#LG.fst.txt is from my ASR of kaldi
f=open(lang_dir / 'LG.fst.txt')
LG = k2.Fsa.from_openfst(f.read(), acceptor=False)
LG = k2.determinize(LG)
LG = k2.connect(LG)
if isinstance(LG.aux_labels, torch.Tensor):
    LG.aux_labels[LG.aux_labels >= first_word_disambig_id] = 0
else:
    LG.aux_labels.values()[
        LG.aux_labels.values() >= first_word_disambig_id] = 0
LG = k2.remove_epsilon(LG)
LG = k2.connect(LG)
LG.aux_labels = k2.ragged.remove_values_eq(LG.aux_labels, 0)
LG = k2.arc_sort(LG)
LG = k2.compose(ctc_topo, LG, inner_labels='phones')
LG = k2.connect(LG)
```
and the problem is: 
    the best path of nnet_output seems ok
    the best path of lattice is completely wrong

and I found:

|LG.arcs_as_tensor().shape   |my exp                    |snowfall_librispeech     |
| -------------------------- | ------------------------ | ----------------------- |
|before compose with ctc_topo|torch.Size([139457381, 4])|torch.Size([32061733, 4])|
|after compose with ctc_topo |torch.Size([8886492, 4])  |torch.Size([99114583, 4])|
|after connect               |torch.Size([169910, 4])   |torch.Size([99114583, 4])|

after compose with ctc_topo, the num of LG.arcs become extreme small, maybe it's where things goes wrong.
any advice?

P.S.:
I do
    LG = k2.determinize(LG)
for a already determinized LG is because
    LG.aux_labels = k2.ragged.remove_values_eq(, 0)
require LG.aux_labels to be type of _k2.RaggedInt

and the aux_labels  of LG  which loaded from LG.fst.txt  is type of torch.tensor
after determinize, it's become _k2.RaggedInt.

**Comment by csukuangfj** [comment]
@Jarvan-Wang 
Could you try the latest k2?
There are several bug fixes about `k2.compose()` and `k2.Fsa.from_openfst`.

**Comment by Jarvan-Wang** [comment]
> 
> 
> @Jarvan-Wang
> Could you try the latest k2?
> There are several bug fixes about `k2.compose()` and `k2.Fsa.from_openfst`.

hmm, no, I dont want to try the latest commit, my partner cannot even ran aishell exp successfuly
this combination of k2 and snowfall seems ok for my librispeech exp

**Comment by csukuangfj** [comment]
> I dont want to try the latest commit, my partner cannot even ran aishell exp successfuly

What kinds of errors do you get? Maybe we can help to fix them.

You don't need to update snowfall. Only the latest k2 is needed.

**Comment by csukuangfj** [comment]
@Jarvan-Wang 
As an alternative, can you try your demo code with the latest k2?
You can create another directory to test the latest k2 without affecting your current settings.

**Comment by Jarvan-Wang** [comment]
> 
> 
> > I dont want to try the latest commit, my partner cannot even ran aishell exp successfuly
> 
> What kinds of errors do you get? Maybe we can help to fix them.
> 
> You don't need to update snowfall. Only the latest k2 is needed.

the testset I decoded is mandarin but the result I got is english results (the L and G support CN and EN both)
pdb stdout is below:
```
(Pdb) print(best_paths.aux_labels)
# in right side of best_path,  43698 is a english word
[ [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ 43698 ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ ] [ -1 ] ]
(Pdb) print(best_paths.labels)
the left side of best_paths, the phone-id from 0 to 96 is all CN phone
tensor([ 0,  0,  0,  0,  0,  0,  0,  0,  0, 94, 52,  0,  0, 11, 86,  0,  0,  0,
         0,  0,  0, 93, 53,  0,  0,  0,  9,  0, 85,  0,  0,  0, 93, 51,  0, 64,
         0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, -1], device='cuda:0',
       dtype=torch.int32)
```
I guess the ctc-LG is completely wrong after composed with ctc_topo

**Comment by Jarvan-Wang** [comment]
> 
> 
> @Jarvan-Wang
> As an alternative, can you try your demo code with the latest k2?
> You can create another directory to test the latest k2 without affecting your current settings.

OK, It's seems the only way I can try now

**Comment by danpovey** [comment]
You are not removing the phonetic  disambig symbols.
    LG.labels[LG.labels >= labels_disambig_id_start] = 0

**Comment by Jarvan-Wang** [comment]
> 
> 
> You are not removing the phonetic disambig symbols.
> LG.labels[LG.labels >= labels_disambig_id_start] = 0

yeah, you are right, I forgot to handle this.
After removing disambig phone symbols.
the num of LG arcs  is as expected.
thx.
